### DISCUSSION TITLE
Nightscout sync is skipping/losing a minority of insulin treatments

### DISCUSSION THREAD

**Discussion by thedewi** [NightscoutFoundation/xDrip#3282] [Bug reports]
Version: 23ccbd1-2023.07.16
Running on: Google Pixel 6 Pro

A seemingly random subset (maybe about 10%?) of insulin treatments are missing from my nightscout instance. I was only keeping 3 months of data in xDrip+ and based on a few spot checks it seems to affect the whole time range.

I've been using the NovoPen insulin pen NFC to enter most treatments. I have no idea if that's a factor, but I mention it just in case.

Obviously the usual troubleshooting steps don't apply, as most of my data does arrive. As far as I know my network is fairly reliable.

Can anyone suggest how I can query the xDrip database to get counts of treatments for certain time ranges, and investigate if there's any specific patterns in the records not being uploaded?

I haven't tried a backfill yet - I'm a bit hesitant to backfill my entire DB until we get more info about the state it's in now.

**Comment by thedewi** [comment]
Note: could be related to #2900? I haven't found any other associated issues yet.

**Comment by thedewi** [comment]
PS. Yesterday was weirdly much worse... it missed 3 in a row (which is why I started investigating), from 2 different scans.

Here are my logs for the 2 scans that never uploaded. These don't seem any different to what's normally logged.

![Screenshot_20240108-163500-3](https://github.com/NightscoutFoundation/xDrip/assets/4726656/8bc64bf3-0def-4a44-b552-e59124b0f9b8)

**Reply by thedewi** [reply]
(Just to clarify: I say "3" because I'm counting non-priming doses. So it's really 6; the priming doses were not uploaded either.)

**Comment by Navid200** [comment]
Please confirm that what is missing in Nightscout is present in xDrip.  
<br/>  
<br/>  


Let's enable detail logs so that hopefully, we get more clues:

Go to 
Settings &#8722;> Less common settings &#8722;> Extra logging settings &#8722;> Extra tags for logging  

Enter the following:  
`UploaderQueue:v`  
  
Please go to the logs and confirm you see some new logs with heading "UploaderQueue" once every 5 minutes.  
![Screenshot_20240108-090027](https://github.com/NightscoutFoundation/xDrip/assets/51497406/d090239b-8c79-4f8c-bf1c-27b0dafab45b)

The next time you have a missing point, let's see the logs.

**Reply by thedewi** [reply]
Thanks for that. I turned that on, but didn't see a problem in the first couple of days, and I won't be using that feature for a while now (I'm trying something with a pump).
